components/movementsensor/gpsutils/cachedData.go

```diff
@@ -224,11 +231,15 @@ func (g *CachedData) ReadSatsInView(ctx context.Context) (int, error) {
 	return g.nmeaData.SatsInView, nil
 }
 
-// ReadSatsInUse returns the number of satellites in use to calculate fix type.
-func (g *CachedData) ReadSatsInUse(ctx context.Context) (int, error) {
-	g.mu.RLock()
-	defer g.mu.RUnlock()
-	return g.nmeaData.SatsInUse, nil
+// GetCommonReadings returns a map including fix value, sats in view and sats in use.
+func (g *CachedData) GetCommonReadings(ctx context.Context) (map[int]interface{}, error) {
+	commonReadings := make(map[int]interface{})
+
+	commonReadings[FixValueKey] = g.nmeaData.FixQuality
+	commonReadings[SatsInViewKey] = g.nmeaData.SatsInView
+	commonReadings[SatsInUseKey] = g.nmeaData.SatsInUse
+
+	return commonReadings, nil
```

Comment:
I don't see any possible way to return a non-nil error. Why return an error at all? I'd get rid of it.